### ISSUE TITLE
iOS Safari: Checkbox doesn't change unless I ignore onFocus

### ISSUE THREAD

**Issue by cpsubrian** [erikras/redux-form#878]
Dealing with the weirdest bug. My redux-form checkboxes aren't changing values on an iPhone in Safari unless I manually ignore the `onFocus` callback in the field props. If I get a moment, I'll try to work up a minimal test case that reproduces this. Desktop browsers don't experience this bug.

When I trace the redux actions, the `redux-form/CHANGE` fires after blur and focus events, with a value = false (the initial state of the checkbox).

Notably, the checkbox in the example 'Simple Form' works fine, so it's something specific to my situation and really could be anything, such as CSS rules, structure of my document, race conditions in my redux flow, etc. Not a clue at the moment.

**Comment by erikras** [comment]
Before I got to your last paragraph, I was testing the simple example on my phone.

Keep me posted.

**Comment by tricoder42** [comment]
I have the same problem in desktop Safari... Example at redux-form website works fine. Setting `onFocus={null}` doesn't solve the problem, but I'm still curios what's the real problem :)

(Note: I'm not reporting the bug or asking for help. Just giving more cases where it might happen)

**Comment by tricoder42** [comment]
It's caused by BrowserSync (at least in my case). Chrome works fine, but Safari don't.

@cpsubrian Could you please confirm if it's also your case?

**Comment by cpsubrian** [comment]
@tricoder42 I was indeed using BrowserSync in development, but I don't recall if I saw the issue in production as well. Will check on it when I have a chance.

**Comment by evanheisler** [comment]
May or may not be related, but I am sorting out an issue where radio buttons are not receiving their "checked" state in production, but works fine locally (I am also using BrowserSync, btw).

Major difference I see is that when it works, there is a FOCUS, BLUR chain of events. When it doesn't work, BLUR is never called. Again, this only occurs in the compiled version of my code, so debugging has been difficult. My input has a custom `onChange` prop, but that doesn't seem to be the issue. Just thought I'd put it out there in case it sparks an idea.

**Comment by evanheisler** [comment]
Quick follow up — I solved my case by manually firing `onBlur` in my custom `onChange` handler (I am submitting responses whenever the input changes)

```
const handleResponse = (choice, dependentOn) => {
    return () => {
      if (dependentOn) props.fields[dependentOn].onChange(null);
      props.fields[props.active].onBlur(choice.body);   // Fire blur WITH input's value
      sendResponse(choice);
    };
};
```

**Comment by dvanoni** [comment]
I saw this same issue, and it did indeed turn out to be caused by Browsersync. My production build (which doesn't use Browsersync) works just fine without having to override any event handlers.

**Comment by mpoisot** [comment]
I'm seeing something similar using redux-form v6.5.0

On Chrome when I toggle the checkbox then focus on another field, there is a BLUR event for the checkbox. But on Safari (macOS and iOS) I don't see the BLUR event. Validations run fine, but "touch=true" is never sent to my CheckBox custom component (presumably due to the lack of BLUR) so my conditional error message doesn't ever show.

I don't know if it's helpful, but I noticed a discrepancy in redux-form's reducer data between the 2 browsers. In Safari my checkbox `tosAgree` never showed up under the `fields` object no matter how much I tickled it.

Chrome:
![screen shot 2017-03-15 at 12 14 46 pm](https://cloud.githubusercontent.com/assets/198049/23961640/8a6e2404-0979-11e7-9fdd-9247054870ad.png)

Safari:
![screen shot 2017-03-15 at 12 15 39 pm](https://cloud.githubusercontent.com/assets/198049/23961639/8a6bab5c-0979-11e7-8b8d-552a41153a00.png)
